confgenerator/otel/conf.go

```diff
@@ -587,31 +582,113 @@ windowsperfcounters/mssql_{{.MSSQLID}}:
           - action: toggle_scalar_data_type
 {{- end -}}`))
 
-type MSSQL struct {
-	MSSQLID            string
+type Receiver interface {
+	Generate() (string, error)
+	GetID() string
+	DefaultPipelineID() string
+	DefaultProcessors() []string
+}
+
+type HostMetrics struct {
+	HostMetricsID      string
 	CollectionInterval string
 }
 
+func (r *HostMetrics) Generate() (string, error) {
+	var builder strings.Builder
+	if err := confTemplate.ExecuteTemplate(&builder, "hostmetrics", r); err != nil {
+		return "", err
+	}
+	return builder.String(), nil
+}
+
+func (r *HostMetrics) GetID() string {
+	return r.HostMetricsID
+}
+
+func (r *HostMetrics) DefaultProcessors() []string {
+	return []string{"agentmetrics/system", "filter/system", "metricstransform/system", "resourcedetection"}
+}
+
+func (r *HostMetrics) DefaultPipelineID() string {
+	return "system"
+}
+
 type IIS struct {
 	IISID              string
 	CollectionInterval string
 }
 
-type HostMetrics struct {
-	HostMetricsID      string
+func (r *IIS) Generate() (string, error) {
```

Comment:
I want to replace this with just returning the otel config structs directly, but happy to defer that to a separate PR. Passing around snippets of YAML text is begging for trouble.

Reply:
Agreed on the "begging for trouble" bit, but let's defer to a separate PR.